### PR TITLE
RBIT function has been implemented. As discussed on issue #7

### PR DESCRIPTION
**Pull request by punoqun**
There are 2 versions, 1 with commented out as a manual version of doing rbit, the other is using Integer.reverseBytes which is not commented out.

### CODE REVIEW

File: src/main/java/com/kasirgalabs/etumulator/visitor/Shift.java [hunk 1]
```diff
@@ -21,5 +21,6 @@
     LSL,
     LSR,
     ROR,
-    RRX;
+    RRX,
+    RBIT;
```
Comment (RootG) [inline src/main/java/com/kasirgalabs/etumulator/visitor/Shift.java]: Needs to be removed.

File: src/main/java/com/kasirgalabs/etumulator/visitor/ShiftVisitor.java [hunk 2]
```diff
@@ -173,6 +173,15 @@ public Void visitRrxs(ProcessorParser.RrxsContext ctx) {
         return null;
     }
 
+    @Override
+    public Void visitRbit(ProcessorParser.RbitContext ctx) {
+        final int NOT_USED = Integer.MAX_VALUE;
+        String destRegister = registerVisitor.visit(ctx.rd());
+        int value = registerFile.getValue(registerVisitor.visit(ctx.rm()));
+        registerFile.setValue(destRegister, shift(value, Shift.RBIT, NOT_USED));
```
Comment (RootG) [inline src/main/java/com/kasirgalabs/etumulator/visitor/ShiftVisitor.java]: Just replace `shift(value, Shift.RBIT, NOT_USED)` with `Integer.reverseBytes(value)`

**Comment by RootG** [comment]
It seemed valid to me accept RBIT is not a shift operation. I have added some inline comments to fix this. 
The method you override from visitor class should be in a class named ReverseVisitor. However, I'll except your pull request if you update your code according to my inline comments.

**Comment by punoqun** [comment]
Oh shoot im on it.

**Pull request merged**